gdx/src/com/badlogic/gdx/utils/JsonSkimmer.java

```diff
@@ -0,0 +1,592 @@
+			1, 1, 0, 0};
+	}
+
+	private static final byte _json_range_lengths[] = init__json_range_lengths_0();
+
+	private static short[] init__json_index_offsets_0 () {
+		return new short[] {0, 0, 11, 14, 16, 19, 28, 34, 40, 43, 54, 62, 70, 79, 81, 90, 93, 96, 105, 108, 111, 113, 116, 119, 130,
+			138, 146, 157, 159, 170, 173, 176, 187, 190, 193, 196, 201, 206, 207};
+	}
+
+	private static final short _json_index_offsets[] = init__json_index_offsets_0();
+
+	private static byte[] init__json_indicies_0 () {
+		return new byte[] {1, 1, 2, 3, 4, 3, 5, 3, 6, 1, 0, 7, 7, 3, 8, 3, 9, 9, 3, 11, 11, 12, 13, 14, 3, 15, 11, 10, 16, 16, 17,
+			18, 16, 3, 19, 19, 20, 21, 19, 3, 22, 22, 3, 21, 21, 24, 3, 25, 3, 26, 3, 27, 21, 23, 28, 29, 29, 28, 30, 31, 32, 3, 33,
+			34, 34, 33, 13, 35, 15, 3, 34, 34, 12, 36, 37, 3, 15, 34, 10, 16, 3, 36, 36, 12, 3, 38, 3, 3, 36, 10, 39, 39, 3, 40, 40,
+			3, 13, 13, 12, 3, 41, 3, 15, 13, 10, 42, 42, 3, 43, 43, 3, 28, 3, 44, 44, 3, 45, 45, 3, 47, 47, 48, 49, 50, 3, 51, 52,
+			53, 47, 46, 54, 55, 55, 54, 56, 57, 58, 3, 59, 60, 60, 59, 49, 61, 52, 3, 60, 60, 48, 62, 63, 3, 51, 52, 53, 60, 46, 54,
+			3, 62, 62, 48, 3, 64, 3, 51, 3, 53, 62, 46, 65, 65, 3, 66, 66, 3, 49, 49, 48, 3, 67, 3, 51, 52, 53, 49, 46, 68, 68, 3,
+			69, 69, 3, 70, 70, 3, 8, 8, 71, 8, 3, 72, 72, 73, 72, 3, 3, 3, 0};
+	}
+
+	private static final byte _json_indicies[] = init__json_indicies_0();
+
+	private static byte[] init__json_trans_targs_0 () {
+		return new byte[] {35, 1, 3, 0, 4, 36, 36, 36, 36, 1, 6, 5, 13, 17, 22, 37, 7, 8, 9, 7, 8, 9, 7, 10, 20, 21, 11, 11, 11, 12,
+			17, 19, 37, 11, 12, 19, 14, 16, 15, 14, 12, 18, 17, 11, 9, 5, 24, 23, 27, 31, 34, 25, 38, 25, 25, 26, 31, 33, 38, 25, 26,
+			33, 28, 30, 29, 28, 26, 32, 31, 25, 23, 2, 36, 2};
+	}
+
+	private static final byte _json_trans_targs[] = init__json_trans_targs_0();
+
+	private static byte[] init__json_trans_actions_0 () {
+		return new byte[] {13, 0, 15, 0, 0, 7, 3, 11, 1, 11, 17, 0, 20, 0, 0, 5, 1, 1, 1, 0, 0, 0, 11, 13, 15, 0, 7, 3, 1, 1, 1, 1,
+			23, 0, 0, 0, 0, 0, 0, 11, 11, 0, 11, 11, 11, 11, 13, 0, 15, 0, 0, 7, 9, 3, 1, 1, 1, 1, 26, 0, 0, 0, 0, 0, 0, 11, 11, 0,
+			11, 11, 11, 1, 0, 0};
+	}
+
+	private static final byte _json_trans_actions[] = init__json_trans_actions_0();
+
+	private static byte[] init__json_eof_actions_0 () {
+		return new byte[] {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+			1, 0, 0, 0};
+	}
+
+	private static final byte _json_eof_actions[] = init__json_eof_actions_0();
+
+	static final int json_start = 1;
+	static final int json_first_final = 35;
+	static final int json_error = 0;
+
+	static final int json_en_object = 5;
+	static final int json_en_array = 23;
+	static final int json_en_main = 1;
+
+	// line 269 "JsonSkimmer.rl"
+
+	private boolean stop;
+
+	/** Causes parsing to stop after the current or next object, array, or value. */
+	public void stop () {
+		stop = true;
```

Comment:
How would one start parsing back up after it stops, if that makes sense at all? Would any starting-back-up make more sense as a new parsing attempt?

Reply:
When you stop you lose all parser state machine state (the locals `cs`, `p`, etc). I can't think of a use case where you want to start/stop parsing. Usually you `stop()` when you've gotten the data you need and don't want to parse the rest at all.

Reply:
Makes sense, seems fine then.